calico_node/readiness/bird/bird.go

```diff
@@ -0,0 +1,275 @@
+package bird
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"net"
+	"reflect"
+	"regexp"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Check for Word_<IP> where every octate is seperated by "_", regardless of IP protocols
+// Example match: "Mesh_192_168_56_101" or "Mesh_fd80_24e2_f998_72d7__2"
+var bgpPeerRegex = regexp.MustCompile(`^(Global|Node|Mesh)_(.+)$`)
+
+// Mapping the BIRD/GoBGP type extracted from the peer name to the display type.
+var bgpTypeMap = map[string]string{
+	"Global": "global",
+	"Mesh":   "node-to-node mesh",
+	"Node":   "node specific",
+}
+
+// Timeout for querying BIRD
+var birdTimeOut = 2 * time.Second
+
+// Expected BIRD protocol table columns
+var birdExpectedHeadings = []string{"name", "proto", "table", "state", "since", "info"}
+
+func CheckGrInProgress(ipv string) error {
+	birdSuffix := ""
+	if ipv == "6" {
+		birdSuffix = "6"
+	}
+
+	// Try connecting to the bird socket in `/var/run/calico/` first to get the data
+	c, err := net.Dial("unix", fmt.Sprintf("/var/run/calico/bird%s.ctl", birdSuffix))
+	if err != nil {
+		// If that fails, try connecting to bird socket in `/var/run/bird` (which is the
+		// default socket location for bird install) for non-containerized installs
+		log.Debugln("Failed to connect to BIRD socket in /var/run/calico, trying /var/run/bird")
```

Comment:
Not sure we need this part, since we're only using this for containerized installs.